cmd/virt-launcher/virt-launcher.go

```diff
@@ -39,40 +40,81 @@ func markReady(readinessFile string) {
 		panic(err)
 	}
 	f.Close()
-	log.Printf("Marked as ready")
+	logging.DefaultLogger().Info().Msgf("Marked as ready")
 }
 
-func createSocket(socketDir string, namespace string, name string) net.Listener {
-	socketPath := isolation.SocketFromNamespaceName(socketDir, namespace, name)
-	if err := os.RemoveAll(socketPath); err != nil {
-		log.Fatal("Could not clean up old socket for cgroup detection", err)
+func createSocket(virtShareDir string, namespace string, name string) net.Listener {
+	sockFile := isolation.SocketFromNamespaceName(virtShareDir, namespace, name)
+
+	err := os.MkdirAll(filepath.Dir(sockFile), 0755)
+	if err != nil {
+		logging.DefaultLogger().Reason(err).Error().Msgf("Could not create directory for socket.")
+		panic(err)
 	}
-	socket, err := net.Listen("unix", isolation.SocketFromNamespaceName(socketDir, namespace, name))
+
+	if err := os.RemoveAll(sockFile); err != nil {
+		logging.DefaultLogger().Reason(err).Error().Msgf("Could not clean up old socket for cgroup detection")
+		panic(err)
+	}
+	socket, err := net.Listen("unix", sockFile)
 
 	if err != nil {
-		log.Fatal("Could not create socket for cgroup detection.", err)
+		logging.DefaultLogger().Reason(err).Error().Msgf("Could not create socket for cgroup detection.")
+		panic(err)
 	}
 
 	return socket
 }
 
 func main() {
 	startTimeout := 0 * time.Second
```

Comment:
So startTimeout is 0 seconds?

Reply:
That's a default value. It doesn't really make sense.